### PR TITLE
RGB-D support ROS2

### PR DESCRIPTION
**Pull request by ShivamPR21**
RGB-D camera support ported and fixed for the latest framework of `ROS2` ecosystem.

| Info | Please fill out this column |
| :-------------------: | :-----------------: |
| OS | Ubuntu 20.04 |
| Ticket(s) this addresses | #3 |
| ROS2 branch | [master](https://raw.githubusercontent.com/ros2/ros2/master/ros2.repos) |
| openvslam | main |

### CODE REVIEW

File: src/openvslam_ros.h [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #include <nav_msgs/msg/odometry.hpp>
 
 #include <opencv2/core/core.hpp>
+#include <sensor_msgs/msg/detail/image__struct.hpp>
```
Comment (ymd-stella) [inline src/openvslam_ros.h]: Wouldn't the following be appropriate?
`#include <sensor_msgs/msg/image.hpp>`

**Review by ymd-stella — COMMENTED** [review]
LGTM!
There was one point that I noticed, so please check it out.

File: src/openvslam_ros.h (hunk 1 above)
Comment (ShivamPR21) [reply src/openvslam_ros.h]: Done with the latest commit, marking as resolved. Thanks for pointing that out.

File: src/openvslam_ros.h [hunk 2]
```diff
@@ -44,11 +45,19 @@ class stereo : public system {
 public:
     stereo(const std::shared_ptr<openvslam::config>& cfg, const std::string& vocab_file_path, const std::string& mask_img_path,
            const bool rectify);
-    void callback(const sensor_msgs::msg::Image::ConstPtr& left, const sensor_msgs::msg::Image::ConstPtr& right);
+    void callback(const sensor_msgs::msg::Image::ConstSharedPtr& left, const sensor_msgs::msg::Image::ConstSharedPtr& right);
```
Comment (AlexeyMerzlyakov) [inline src/openvslam_ros.h:48]: Did you perform a run-time verification of this (I mean Gazebo TB3 simulation, synthetic or ROSBAG topics publishing or something else)?

**Review by AlexeyMerzlyakov — APPROVED** [review]
Please check one question about callbacks verification. Otherwise, looks good.

**Review by SteveMacenski — APPROVED** [review]
LGTM

**Pull request merged**